lib/rubygems/specification.rb

```diff
@@ -2693,7 +2693,7 @@ def version=(version)
     @version = Gem::Version.create(version)
     # skip to set required_ruby_version when pre-released rubygems.
     # It caused to raise CircularDependencyError
-    if @version.prerelease? && @name.strip != "rubygems"
+    if @version.prerelease? && (@name.nil? || @name.strip != "rubygems")
```

Comment:
Should we do a casecmp here in case of "RuByGeMs"?

Reply:
Maybe it's a good idea, because `bundler` seems to have changed the casing of this string, actually :point_right: https://github.com/bundler/bundler/commit/fcc4b607379b25f607ce8d003664902f58fefd23.